editor/components/post-sticky/test/index.js

```diff
@@ -9,54 +9,51 @@ import { shallow } from 'enzyme';
 import { PostStickyCheck } from '../check';
 
 describe( 'PostSticky', () => {
-	const user = {
-		data: {
-			post_type_capabilities: {
-				edit_others_posts: true,
-				publish_posts: true,
-			},
-		},
-	};
-
 	it( 'should not render anything if the post type is not "post"', () => {
 		const wrapper = shallow(
-			<PostStickyCheck postType="page" user={ user }>
+			<PostStickyCheck postType="page" post={ {
+				_links: {
+					self: {
+						href: 'https://w.org/wp-json/wp/v2/posts/5',
+					},
+				},
+				title: 'Not a stickyable post',
+			} }>
 				Can Toggle Sticky
 			</PostStickyCheck>
 		);
 		expect( wrapper.type() ).toBe( null );
 	} );
 
-	it( 'should not render anything if the user doesn\'t have the right capabilities', () => {
-		let wrapper = shallow(
-			<PostStickyCheck postType="post" user={ {} }>
-				Can Toggle Sticky
-			</PostStickyCheck>
-		);
-		expect( wrapper.type() ).toBe( null );
-
-		wrapper = shallow(
-			<PostStickyCheck postType="post" user={
-				{ data: { post_type_capabilities: { edit_others_posts: false, publish_posts: true } } }
-			}>
-				Can Toggle Sticky
-			</PostStickyCheck>
-		);
-		expect( wrapper.type() ).toBe( null );
-
-		wrapper = shallow(
-			<PostStickyCheck postType="post" user={
-				{ data: { post_type_capabilities: { edit_others_posts: true, publish_posts: false } } }
-			}>
+	it( 'should not render anything if post doesn\'t support stickying', () => {
+		const wrapper = shallow(
+			<PostStickyCheck postType="post" post={ {
+				_links: {
+					self: {
+						href: 'https://w.org/wp-json/wp/v2/posts/5',
+					},
+				},
+				title: 'Not a stickyable post',
+			} }>
 				Can Toggle Sticky
 			</PostStickyCheck>
 		);
 		expect( wrapper.type() ).toBe( null );
 	} );
 
-	it( 'should render if the user has the correct capability', () => {
+	it( 'should render if the post supports stickying', () => {
 		const wrapper = shallow(
-			<PostStickyCheck postType="post" user={ user }>
+			<PostStickyCheck postType="post" post={ {
+				_links: {
+					self: {
+						href: 'https://w.org/wp-json/wp/v2/posts/5',
+					},
+					'wp:action-sticky': {
+						href: 'https://w.org/wp-json/wp/v2/posts/5',
+					},
```

Comment:
Once again the test works as-is because of what we're testing for, but all of the keys on the `_links` object hold arrays of objects, instead of plain objects (e.g. `wp:action-sticky': [   { /* ... */ }   ]`).

Reply:
67f0f9bc87b5bd9c28f1269fb3fe0098490f995b